src/main/java/org/example/springintro/exception/CustomGlobalExceptionHandler.java

```diff
@@ -42,4 +43,22 @@ private String getErrorMessage(ObjectError e) {
         }
         return e.getDefaultMessage();
     }
+
+    @ExceptionHandler(RegistrationException.class)
+    public ResponseEntity<String> handleRegistrationException(RegistrationException e) {
+        return new ResponseEntity<>(e.getMessage(), HttpStatus.CONFLICT);
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    public ResponseEntity<String> handleEntityNotFoundException(EntityNotFoundException e) {
+        return new ResponseEntity<>(e.getMessage(), HttpStatus.NOT_FOUND);
+    }
+
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<String> handleGenericException(Exception e) {
+        return new ResponseEntity<>(
+                "An error occurred: " + e.getMessage(),
+                HttpStatus.INTERNAL_SERVER_ERROR
+        );
+    }
```

Comment:
it doesn't apply to tests, but it's too general exception to catch. Don't do that ;)